#! /bin/sh /usr/share/dpatch/dpatch-run
## 10_empty_-L_flag.dpatch by Fabian Greffrath <fabian+debian@greffrath.com>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Fix libFLAC.m4 may set empty -L flag when configure is called without
## DP: arguments, provided of course that the target package's configure script  
## DP: is properly re-generated against the fixed libFLAC.m4

@DPATCH@
diff -urNad '--exclude=CVS' '--exclude=.svn' '--exclude=.git' '--exclude=.arch' '--exclude=.hg' '--exclude=_darcs' '--exclude=.bzr' flac~/src/libFLAC/libFLAC.m4 flac/src/libFLAC/libFLAC.m4
--- flac~/src/libFLAC/libFLAC.m4	2010-06-15 13:14:45.000000000 +0200
+++ flac/src/libFLAC/libFLAC.m4	2010-06-15 13:29:17.000000000 +0200
@@ -14,14 +14,18 @@
 AC_ARG_ENABLE(libFLACtest, [  --disable-libFLACtest       Do not try to compile and run a test libFLAC program],, enable_libFLACtest=yes)
 
   if test "x$libFLAC_libraries" != "x" ; then
-    LIBFLAC_LIBDIR="$libFLAC_libraries"
+    LIBFLAC_LIBS="-L$libFLAC_libraries"
+  elif test "x$libFLAC_prefix" = "xno" || test "x$libFLAC_prefix" = "xyes" ; then
+    LIBFLAC_LIBS=""
   elif test "x$libFLAC_prefix" != "x" ; then
-    LIBFLAC_LIBDIR="$libFLAC_prefix/lib"
-  elif test "x$prefix" != "xNONE" ; then
-    LIBFLAC_LIBDIR="$libdir"
+    LIBFLAC_LIBS="-L$libFLAC_prefix/lib"
+  elif test "x$prefix" != "xNONE"; then
+    LIBFLAC_LIBS="-L$prefix/lib"
   fi
 
-  LIBFLAC_LIBS="-L$LIBFLAC_LIBDIR -lFLAC $OGG_LIBS -lm"
+  if test "x$libFLAC_prefix" != "xno" ; then
+    LIBFLAC_LIBS="$LIBFLAC_LIBS -lFLAC $OGG_LIBS -lm"
+  fi
 
   if test "x$libFLAC_includes" != "x" ; then
     LIBFLAC_CFLAGS="-I$libFLAC_includes"
